src/generators/java/JavaGenerator.ts

```diff
@@ -48,4 +50,46 @@ export class JavaGenerator extends AbstractGenerator<JavaOptions> {
     const result = await renderer.runSelfPreset();
     return RenderOutput.toRenderOutput({result, dependencies: renderer.dependencies});
   }
+
+  /**
+   * Generates all the models to an output directory. 
+   * 
+   * This function is invasive, as it overwrite any existing files with the same name as the model.
+   * 
+   * @param input
+   * @param outputDirectory where you want the models generated to
+   * @param packageName for the models
+   */
+  public async generateToFile(input: Record<string, unknown> | CommonInputModel, outputDirectory: string, packageName: string): Promise<OutputModel[]> {
+    const generatedModels = await this.generateFullOutput(input, packageName);
+    for (const outputModel of generatedModels) {
+      const outputFilePath = path.resolve(outputDirectory, `${FormatHelpers.toPascalCase(outputModel.model.$id || 'undefined')}.java`);
+      await FileHelpers.writeToFile(outputModel.result, outputFilePath);
+    }
+    return Promise.resolve(generatedModels);
```

Comment:
Why do you call `Promise.resolve` even function is async? You can just return `generatedModels` without wrapping with Promise.

Reply:
Not sure why, but had some problems with it, so just opted for a more explicit resolvement 😅 Solved!